pkg/kubelet/dockertools/docker.go

```diff
@@ -146,23 +146,23 @@ func filterHTTPError(err error, image string) error {
 
 func (p dockerPuller) Pull(image string, secrets []api.Secret) error {
 	// If no tag was specified, use the default "latest".
-	repoToPull, tag := parsers.ParseImageName(image)
-
-	opts := docker.PullImageOptions{
-		Repository: repoToPull,
-		Tag:        tag,
-	}
+	imageID, tag := parsers.ParseImageName(image)
 
 	keyring, err := credentialprovider.MakeDockerKeyring(secrets, p.keyring)
 	if err != nil {
 		return err
 	}
 
-	creds, haveCredentials := keyring.Lookup(repoToPull)
+	opts := dockertypes.ImagePullOptions{
+		Tag:     tag,
+		ImageID: imageID,
+	}
+
+	creds, haveCredentials := keyring.Lookup(imageID)
 	if !haveCredentials {
 		glog.V(1).Infof("Pulling image %s without credentials", image)
 
-		err := p.client.PullImage(opts, docker.AuthConfiguration{})
+		err := p.client.PullImage(imageID, dockertypes.AuthConfig{}, opts)
```

Comment:
Do we have to pass imageID both as an argument and as a filed in the options? The client should be able to write the imageID to the options.